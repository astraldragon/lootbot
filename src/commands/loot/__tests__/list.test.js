jest.mock("discord-akairo");

jest.mock("../../../models", () => ({
  Loot: { findAll: jest.fn("findAll") },
  Tier: { findAll: jest.fn("findAll") }
}));

const { Loot, Tier } = require("../../../models");
const { Command } = require("discord-akairo");
const ListCommand = require("../list");

describe("commands/loot/list", () => {
  beforeEach(() => {
    this.command = new ListCommand();

    this.msg = {
      guild: { id: "1234" },
      channel: {
        send: jest.fn("send").mockReturnValue(Promise.resolve())
      }
    };
  });

  it("should configure properly", () => {
    expect(Command).toMatchSnapshot();
  });

  it("should fetch the loot for the guild", () => {
    this.command.exec(this.msg);

    expect(Tier.findAll).toHaveBeenCalledWith({
      include: [{ model: Loot }],
      where: { guild: this.msg.guild.id },
      order: [["weight", "DESC"], ["Loots", "name", "ASC"]]
    });
  });

  it("should notify when no loot is found for the guild", async () => {
    Tier.findAll.mockReturnValue([]);
    await this.command.exec(this.msg);
    expect(this.msg.channel.send).toHaveBeenCalledWith("No loot found.");
  });

  it("should list the loot", async () => {
    Tier.findAll.mockReturnValue([
      {
        name: "Common",
        Loots: []
      },
      {
        name: "Uncommon",
        Loots: [{ name: "Test Loot 1" }, { name: "Test Loot 2" }]
      }
    ]);

    await this.command.exec(this.msg);
    expect(this.msg.channel.send).toMatchSnapshot();
  });

  it("should notify if any errors occur", async () => {
    Tier.findAll.mockImplementation(() => {
      throw new Error("test");
    });

    await this.command.exec(this.msg);

    expect(this.msg.channel.send).toHaveBeenCalledWith(
      "An error occurred listing loot."
    );
  });
});
