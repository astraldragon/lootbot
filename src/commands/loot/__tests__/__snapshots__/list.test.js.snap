// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commands/loot/list should configure properly 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "loot-list",
      Object {
        "aliases": Array [
          "loot-list",
          "ll",
        ],
        "category": "Loot",
        "channelRestriction": "guild",
        "description": Object {
          "content": "List all loot in the lootbox",
          "examples": Array [
            "loot-list",
          ],
        },
      },
    ],
  ],
}
`;

exports[`commands/loot/list should list the loot 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "__**Common**__
N/A

__**Uncommon**__
Test Loot 1
Test Loot 2

",
    ],
  ],
}
`;
